src/js/editor/editor.js

```diff
@@ -287,9 +287,14 @@ class Editor {
     this.trigger('update');
   }
 
-  selectSections(sections) {
-    this.cursor.selectSections(sections);
-    this.reportSelection();
+  selectSections(sections=[]) {
+    if (sections.length) {
+      this.cursor.selectSections(sections);
+      this.reportSelection();
+    } else {
+      this.cursor.clearSelection();
+      this.reportNoSelection();
```

Comment:
Should reporting of selection state be something the cursor does? It can fire a callback and that can trigger the complex callbacks on the editor